app/src/main/java/com/zeapo/pwdstore/SearchableRepositoryViewModel.kt

```diff
@@ -56,31 +57,8 @@ else
     PasswordItem.newCategory(name, this, PasswordRepository.getRepositoryDirectory())
 
 private fun PasswordItem.fuzzyMatch(filter: String): Int {
-    var i = 0
-    var j = 0
-    var score = 0
-    var bonus = 0
-    var bonusIncrement = 0
-
-    val toMatch = longName
-
-    while (i < filter.length && j < toMatch.length) {
-        when {
-            filter[i].isWhitespace() -> i++
-            filter[i].toLowerCase() == toMatch[j].toLowerCase() -> {
-                i++
-                bonusIncrement += 1
-                bonus += bonusIncrement
-                score += bonus
-            }
-            else -> {
-                bonus = 0
-                bonusIncrement = 0
-            }
-        }
-        j++
-    }
-    return if (i == filter.length) score else 0
+    val result = FuzzySearch.partialRatio(filter, longName)
+    return if (result < 30) 0 else result
```

Comment:
This `30` is completely arbitrary and should be dropped. I'm too tired right now to fully grok the algorithm going on above, but this does appear to work in my testing. The only problem is that where the existing algorithm correctly clamped return value to drop matches that were too far off, this lists everything.

Reply:
I will need to think about this some more. Fuzzywuzzy does list everything because it also accounts for deleted characters, whereas our old fuzzy search required the filter to be a subsequence (not necessarily substring) of the matched term. Maybe fuzzywuzzy is not what we want after all... I'll look into what browserpass is doing as I quite like their search.

Reply:
Intellij also doesn't account for deletions. Maybe we should keep the original algorithm instead and tweak it a bit? Do we just want the matches where the filter is a substring (not just a subsequence) to sort first?

Reply:
Yeah prioritising substrings will probably alleviate most concerns.